### PR TITLE
Cleanup and improve Model metadata/capabilities

### PR DESCRIPTION
**Pull request by Luthaf**
This update `ModelCapabilities` to add the `interaction_range` and `supported_devices`; and add a `ModelMetadata` class to record name/description/authors/references associated with a model.

Part of #430 

As always, this is best reviewed commit by commit!

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [x] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [x] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://metatensor--526.org.readthedocs.build/en/526/

### CODE REVIEW

**Comment by frostedoyster** [comment]
Do we have an interface now?

**Comment by Luthaf** [comment]
The interface is only here for documentation purpose, since the TorchScript compiler does not check against it.

**Review by frostedoyster — APPROVED** [review]
LGTM

File: metatensor-torch/include/metatensor/torch/atomistic/model.hpp [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    /// List of authors for this model
+    std::vector<std::string> authors;
+
+    /// References for this model. The top level dict can have three keys:
+    ///
+    /// - "implementation": for reference to software used in the implementation
+    ///   of the model
+    /// - "architecture": for reference that introduced the general architecture
+    ///   used by this model
+    /// - "model": for reference specific to this exact model
```
Comment (PicoCentauri) [inline metatensor-torch/include/metatensor/torch/atomistic/model.hpp:227]: For me this is clear but maybe we give and for example?

```suggestion
    /// - "implementation": for reference to software and libraries used in the implementation
    ///   of the model i.e. pytorch: https://dl.acm.org/doi/10.5555/3454287.3455008
    /// - "architecture": for reference that introduced the general architecture
    ///   used by this model
    /// - "model": for reference specific to this exact model.
```

File: metatensor-torch/src/atomistic/model.cpp [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+                    // line_length - 1
+                    view = view.substr(i + 1);
+                    first_line = false;
+                    break;
+                }
+            }
+
+            if (!word_found) {
+                // this is only hit if a single word takes a full line.
+                throw std::runtime_error("your words are too big, make them smaller");
```
Comment (PicoCentauri) [inline metatensor-torch/src/atomistic/model.cpp:523]:
```suggestion
                throw std::runtime_error("Your words are too long to be wrapped. Try making them shorter");
```

File: metatensor-torch/src/atomistic/model.cpp [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+            }
+        }
+    }
+}
+
+
+std::string ModelMetadataHolder::print() const {
```
Comment (PicoCentauri) [inline metatensor-torch/src/atomistic/model.cpp:530]: fancy cpp function!!!

**Pull request merged**